src/main.rs

```diff
@@ -125,7 +127,20 @@ fn main() {
             .subcommand(
                 SubCommand::with_name("explain").about("Explains the currently showing modules"),
             )
-            .get_matches();
+            .subcommand(
+                SubCommand::with_name("completions")
+                    .about("Generate starship shell completions for your shell to stdout")
+                    .arg(
+                        Arg::with_name("shell")
+                            .takes_value(true)
+                            .possible_values(&Shell::variants())
+                            .help("the shell to generate completions for")
+                            .value_name("SHELL")
+                            .required(true),
```

Comment:
We could get around this requirement by reading the `STARSHIP_SHELL` env variable by default, and let the user override by explicitly specifying.

Reply:
Is that actually preferable? It doesn't seem like the requirement is bad, and it does seem like the only other place it's used is in `custom command`, where it's used for something unrelated (specifying the shell to run the custom command in). Typically that would be `sh`, even if I'm using something like `fish` as my interactive shell.

Reply:
It's used in a few places, including our bug-report mechanism. I'd be in favor of having the default behavior be to generate shell completions for the current shell.

Reply:
I agree, but its use in "custom command" implies that `STARSHIP_SHELL` isn't necessarily your current shell? Since it's used to specify the shell that those commands are run in, which is nearly always going to be something like `sh` instead of `fish` or whatever

Reply:
Custom commands accept a `shell` value, which will fallback to `STARSHIP_SHELL` if unset. That is because we set `STARSHIP_SHELL` in the Starship initialization script; we don't expect it to be set by the user. It is assumed to be the currently running shell.

Reply:
Ahh, I see that. I wasn't looking at the init scripts. Very well, will update.

Reply:
Probably this means that `starship init` should be updated too, right? For consistency?

Reply:
Fixed

Reply:
I confirmed that the `env()` and `required()` interaction is correct; the argument is accepted at the command line or via env var. It's an error if both are absent, or if the input doesn't match one of the possible choices.